### PR TITLE
files, template: allow setting registry mirrors

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/11736

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: files/conf/docker-daemon.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "registry-mirrors": [
+        {{- range $i, $e := .RegistryMirrors }}
```
Comment (corest) [inline files/conf/docker-daemon.json:3]: download (9)
 is it only for me formatting looks like this?

Comment (kopiczko) [reply files/conf/docker-daemon.json:3]: It's for everyone. This is probably because it isn't a valid JSON before templating.

**Comment by kopiczko** [comment]
Sorry for the noise. I'm turning this into draft and let you know when I think it's ready.

**Comment by kopiczko** [comment]
Tested here https://github.com/giantswarm/aws-operator/pull/2572

**Pull request merged**